src/uwtools/set_config.py

```diff
@@ -94,7 +117,13 @@ def create_config_obj(argv):
         config_obj.update_values(user_config_obj)
 
     if user_args.outfile:
-        config_obj.dump_file(user_args.outfile)
+        outfile_type = user_args.out_file_type or pathlib.Path(user_args.outfile).suffix
+        if outfile_type == "field_table":
+            out_object = config.FieldTableConfig()
+            out_object.update(config_obj)
+        else:
```

Comment:
When we chatted about this, we'd mentioned that this could be converted to any output object type. It would be nice to apply the same sort of logic as above to determine what sort of output config object should be created, not just handling the Field Table case.

That gives us a lot of flexibility, including what we need for the Field Table.

Reply:
It is a little tough to say, as the FieldTableConfig just handles field tables, but I think generalizing it here and adjusting as we add more table types later. There are some complexities in the logic, since the bash process reads INI and outputs SH, but I think I can resolve it fairly quickly

Reply:
Got it - fixing by standardizing expected infile_types with multiple options, so the config should now only trigger on intentional changes

Reply:
Fixed with improved logic